server/src/service.rs

```diff
@@ -107,10 +107,38 @@ impl Service {
     #[zbus(out_args("secrets"))]
     pub async fn get_secrets(
         &self,
-        _items: Vec<OwnedObjectPath>,
-        _session: ObjectPath<'_>,
+        items: Vec<OwnedObjectPath>,
+        session: OwnedObjectPath,
     ) -> Result<HashMap<OwnedObjectPath, SecretInner>, ServiceError> {
-        todo!()
+        let mut secrets: HashMap<OwnedObjectPath, SecretInner> = HashMap::new();
+        let collections = self.collections.lock().await;
+
+        'outer: for collection in collections.iter() {
+            for item in &items {
+                if let Some(item) = collection.item_inner(item).await {
```

Comment:
Instead of adding this function, you should make the search_items_inner function from the that other PR takes a search condition function, that way you can use it both for comparing attributes or item paths.

Reply:
Thanks! Sorry for the late reply :)
Yeah it make sense. But, the returns are different. For this case we're returning a single Item based on the objectpath and in SearchItems we're returning multiple Items based on attributes.
So, it is bad to have two methods: `search_inner_items(attributes)` and `inner_item(objectpath)`?

Reply:
The name is just so weird. Maybe find_item_with_path or so?

Reply:
yeah, I agree :) will update it.